packages/flutter/lib/src/material/reorderable_list.dart

```diff
@@ -54,6 +54,45 @@ typedef ReorderCallback = void Function(int oldIndex, int newIndex);
 /// All [children] must have a key.
 ///
 /// {@youtube 560 315 https://www.youtube.com/watch?v=3fB1mxOsqJE}
+///
+/// This sample shows by dragging the user can reorder the items of the list.
+/// list of Numbers will be used to display items in listview.
+/// [List.generate]it is constructor which  is used to form a list of values.
+/// [onReorder] This parameter is compulsory. It will be called when list child
+/// is moved to a new position.
+///
+/// {@tool dartpad --template=stateful_widget_scaffold}
+///
+/// ```dart
+/// List<String> _list = List.generate(5, (i) => "${i}");
+///
+/// Widget build(BuildContext context){
+///   return ReorderableListView(
+///     padding : const EdgeInsets.symmetric(horizontal:40),
+///     children: _list.map((String string) => ListTile(
+///       key: Key(_list[_list.indexOf(string)]),
```

Comment:
It's probably better to base the key off of `_list.indexOf(string)` instead of the value at that location. This section could also be simplified a little by using a *collection for*:

```dart
      children: [
        for (var i = 0; i < _list.length; i++)
          ListTile(key: Key('$i'), title: Text(_list[i])),
      ],
```